### PR TITLE
Add support for Client Customization and app icons for GAE

### PR DESCRIPTION
**Pull request by selfcommit**
Include support for [Client Customization](https://github.com/munki/munki/wiki/Client-Customization) and [App Icons](https://github.com/munki/munki/wiki/Product-Icons) in GAE.

### CODE REVIEW

**Comment by selfcommit** [comment]
After thinking more about this, it might be better to symlink the client_resources and icons folders only if they exists in PWD/src.

**Comment by selfcommit** [comment]
I'm happy with this now.  Correct me if the /src/ folder is the wrong place for housing install specific files.  ping @juofmgoblue

**Comment by juofmgoblue** [comment]
Dan, thanks for the pull request.  Apologies for the delayed response; you caught me while on vacation and now I'm busy catching up. 

In general I have two concerns:

1) given these are "static_dir" lacking "login: required" in app.yaml, the resources within can be fetched by anyone that knows how to address them, without any auth.  Ideally we'd use Simian's auth protocol to protect them, but I don't know of a way to do that with static_dir.   Thus, the secure way of hosting these would be via a custom handler which enforces the custom auth dance (or a Simian Auth1Token), and serves the resource only if authenticated.  However, that's non-trivial, as then the resources probably shouldn't be stored on disk as static content, but rather stored in Datastore or Blobstore.  This is the reason we haven't built such support yet.

I think I'm OK with this until we get around to a more elegant approach, so long as we have clear documentation that the resources placed in this dirs are not secured/private, and are publicly available to anyone that can guess their names.   Would you still want to use this as is, knowing these details?


2) /src/ is probably not a great location for this, given that should typically always match the Git repo, unmodified.  the only other path users have to edit is etc/, which also doesn't fit.  maybe we should create a new top-level "gae_resources" dir, or something similar?  I don't feel strongly here; open to suggestions or pushback.

**Comment by selfcommit** [comment]
Thanks for your reply Justin, hoped you enjoyed vacation.  I'm interested in contributing back to the project regularly and thought this would be a good "get to know each other" pull request.

1) You raise a valid point about the static content.  I don't know if, out of the box, GAE has any kind of rate limiting to prevent abuse.  Icons are small, but the example zip file would be ripe for abuse.
  Maybe remove the example .zip but leave enough breadcrumbs for those that want to set it up?

2) gae_resource seems like a good choice.  I'll update the pull request to match that.

Maybe down the line a useful feature would be to have the simian_client (or another script) call Munki's makepkginfo and then pass icons / relevant package info directly the GAE instance in a way that allowed for Datastore retrieval.

**Comment by juofmgoblue** [comment]
Proposal:
  client_resources: How about leave it there, but comment out the app.yaml handler so it's "inactive", then clarify how it's not safe for security and privacy of your site customizations, and anyone, even without auth, can download the file very easily since the address is widely known.  Then we can leave it up to the individual to keep their customizations non-confidential if they still choose to use the feature for minor adjustments.  All they'd have to do is uncomment a few lines, after making a conscious decision.

  icons:  probably OK exposed to the public, but still add a comment to the icons handler with a similar but more lax warning.  Still warn against exposing confidential data.  The filename would need to be known, etc.  But leave it enabled by default.

**Comment by selfcommit** [comment]
Agreed and Done.

File: gae_resources/client_resources/customize_client.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Zip files for client customization go here.
```
Comment (juofmgoblue) [inline gae_resources/client_resources/customize_client.md]: Add warnings here.

File: gae_resources/icons/icons.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Place custom icons for apps in this directory.
```
Comment (juofmgoblue) [inline gae_resources/icons/icons.md]: Add warnings here.

**Comment by juofmgoblue** [comment]
A few quick nits/comments, then I'll merge.  I just really want to explicitly call out the concern wherever possible, as most admins are not likely to notice the lack of "login: admin|required", let alone understand the implications.  Thanks again for the pull request!

**Comment by selfcommit** [comment]
Should be all set now.  Thanks for your review time.

**Pull request merged**